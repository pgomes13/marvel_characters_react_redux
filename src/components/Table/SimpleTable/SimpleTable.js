import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { SimpleTableStyles } from './SimpleTable.styles';
import Button from "@material-ui/core/Button/Button";
import {Link} from "react-router-dom";

/**
 * The SimpleTable component consisting of the material-ui Table Component
 * @param {String} title - the title prop
 * @param {String} pathname - the pathname prop
 * @param {Array} items - the items array prop
 * @param {Array} labels - the labels array prop
 * @param {Array} identifiers - the identifiers array prop
 * @param {Function} setAction - the setAction function
 * @param {Function} dispatch - the dispatch function prop
 * @returns {Node} - the SimpleCard component
 */
const _SimpleTable = ({ classes, labels, items, identifiers, pathname, setAction, dispatch }) => {
    const setClickAction = (url) => {
        return dispatch(
            setAction({
                topic: null,
                id: null,
                subTopic: null,
                url: url
            })
        );
    };

    return (
        <Paper className={classes.root}>
            <Table className={classes.table}>
                <TableHead>
                    <TableRow>
                        {labels.map((label, key) => (
                            <TableCell key={key}>{label}</TableCell>
                        ))}
                    </TableRow>
                </TableHead>
                <TableBody>
                    { items.map(((item, key) => (
                        <TableRow key={key}>
                            {identifiers.map((identifier, key) => (
                                identifier === 'resourceURI' ?
                                    <TableCell component="th" scope="row" key={key}>
                                        <Link to={pathname} key={key} onClick={() => setClickAction(item[identifier])}>
                                            <Button size="small" color="primary" key={key}>View</Button>
                                        </Link>
                                    </TableCell>
                                :
                                    <TableCell component="th" scope="row" key={key}>{item[identifier]}</TableCell>
                            ))}
                        </TableRow>
                    )))}
                </TableBody>
            </Table>
        </Paper>
    );
};

_SimpleTable.propTypes = {
    classes: PropTypes.object.isRequired,
    labels: PropTypes.array.isRequired,
    items: PropTypes.array.isRequired,
    pathname: PropTypes.string.isRequired,
    identifiers: PropTypes.array.isRequired,
    isValueUrl: PropTypes.bool,
    setAction: PropTypes.func,
    dispatch: PropTypes.func
};

const SimpleTable = withStyles(SimpleTableStyles)(_SimpleTable);

export { _SimpleTable, SimpleTable };
