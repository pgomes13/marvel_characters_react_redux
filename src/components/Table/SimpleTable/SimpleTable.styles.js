const SimpleTableStyles = theme => ({
    root: {
        width: '100%',
        overflowX: 'auto',
    },
    table: {
        minWidth: 700,
    }
});

export { SimpleTableStyles };