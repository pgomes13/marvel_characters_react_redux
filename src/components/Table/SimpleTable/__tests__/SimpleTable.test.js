import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { _SimpleTable } from '../SimpleTable';

const Fn = jest.fn()

describe('SimpleTable', () => {
    const props = {
        classes: {},
        imageUrl: 'http://test.test.com',
        title: 'test',
        pathname: '/test/test/test',
        setAction: Fn,
        dispatch: Fn,
        items: [{
            test1: 'test1',
            test2: 'test2'
        }],
        labels: ['test', 'test'],
        identifiers: ['test1', 'test2']
    };



    it('component renders correctly with props', () => {
        const wrapper = shallow(<_SimpleTable {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});