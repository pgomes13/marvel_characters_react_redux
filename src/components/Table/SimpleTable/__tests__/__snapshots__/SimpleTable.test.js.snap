// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleTable component renders correctly with props 1`] = `
<WithStyles(Paper)>
  <WithStyles(Table)>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)
          key="0"
        >
          test
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          key="1"
        >
          test
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)
        key="0"
      >
        <WithStyles(TableCell)
          component="th"
          key="0"
          scope="row"
        >
          test1
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          component="th"
          key="1"
          scope="row"
        >
          test2
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</WithStyles(Paper)>
`;
